crates/astria-composer/src/searcher/mod.rs

```diff
@@ -216,111 +156,96 @@ impl Searcher {
-                .await
-                .wrap_err("failed to submit transaction to sequencer")?;
-            if rsp.code.is_err() {
-                bail!(
-                    "submitting transaction to sequencer returned with error code; code: \
-                     `{code}`; log: `{log}`; hash: `{hash}`",
-                    code = rsp.code.value(),
-                    log = rsp.log,
-                    hash = rsp.hash,
-                );
-            }
-            Ok(())
-        });
+    async fn handle_bundle_execution(&self, bundle: Vec<Action>) {
+        // send bundle to executor
+        if let Err(e) = self.executor_tx.send(bundle).await {
+            error!(
+                error.message = %e,
+                error.cause_chain = ?e,
+                "failed to send bundle to executor",
+            );
+        }
     }
 
     /// Starts the searcher and runs it until failure
     pub(super) async fn run(mut self) -> eyre::Result<()> {
         self.spawn_collectors();
+        let mut executor_handle = tokio::spawn(
+            self.executor
+                .take()
+                .expect("executor should only be run once")
+                .run_until_stopped(),
+        );
+
         let wait_for_collectors = self.wait_for_collectors();
-        let wait_for_seq = self.wait_for_sequencer(5, Duration::from_secs(5), 2.0);
-        match tokio::try_join!(wait_for_collectors, wait_for_seq) {
+        let wait_for_executor = self.wait_for_executor();
+        match tokio::try_join!(wait_for_collectors, wait_for_executor) {
             Ok(((), ())) => {}
             Err(err) => return Err(err).wrap_err("failed to start searcher"),
         }
-        // set initial sequencer nonce
-        let address = Address::from_verification_key(self.sequencer_key.verification_key());
-        let nonce_response = self
-            .sequencer_client
-            .inner
-            .get_latest_nonce(address.0)
-            .await
-            .wrap_err("failed to query sequencer for nonce")?;
-        self.sequencer_nonce
-            .store(nonce_response.nonce, Ordering::Relaxed);
 
         loop {
             select!(
                 // serialize and sign sequencer tx for incoming pending rollup txs
-                Some(rollup_tx) = self.new_transactions.recv() => self.handle_pending_tx(rollup_tx),
+                Some(rollup_tx) = self.new_transactions.recv() => self.bundle_pending_tx(rollup_tx),
 
                 // submit signed sequencer txs to sequencer
                 Some(join_result) = self.conversion_tasks.join_next(), if !self.conversion_tasks.is_empty() => {
                     match join_result {
-                        Ok(Ok(signed_tx)) => self.handle_signed_tx(signed_tx),
-                        Ok(Err(e)) => warn!(error.message = %e, error.cause_chain = ?e, "failed to sign sequencer transaction"),
+                        Ok(bundle) => self.handle_bundle_execution(bundle).await,
```

Comment:
This now effectively blocks the entire loop

Reply:
see #409